README.txt
In order to add a new widget to t3blog, you need to:
- create a widget => see files in widgets/trend/
- create a hook to tell t3blog about widgets => see widgets/class.tx_t3blogwidgetdemo_getwidgets.php
- add TypoScript Setup => see static/setup.txt

That's it!

ChangeLog
2010-05-27  Dmitry Dulepov <dmitry.dulepov@gmail.com>

	* Released version 0.1.0


widgets/class.tx_t3blogwidgetdemo_getwidgets.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2010 Dmitry Dulepov (dmitry.dulepov@gmail.com)
*  All rights reserved
*
*  This script is part of the Typo3 project. The Typo3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * $Id$
 *
 */

/**
  * This class is a hook to return widget information to t3blog.
  *
  * @author Dmitry Dulepov <dmitry.dulepov@gmail.com>
  * @package TYPO3
  * @subpackage tx_t3blogwidgetdemo_getwidgets
  */
class tx_t3blogwidgetdemo_getwidgets {

	/**
	 * Provides information about widgets in this extension
	 *
	 * @param array $unusedParams
	 * @return array
	 */
	public function getWidgets(array $unusedParams) {
		return array(
			'tx_t3blogwidgetdemo_trend' => 'EXT:t3blog_widgetdemo/widgets/trend/'
		);
	}

}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/t3blog_widgetdemo/widgets/class.tx_t3blogwidgetdemo_getwidgets.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/t3blog_widgetdemo/widgets/class.tx_t3blogwidgetdemo_getwidgets.php']);
}

?>

widgets/trend/class.tx_t3blogwidgetdemo_trend.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2010 Dmitry Dulepov (dmitry.dulepov@gmail.com)
*  All rights reserved
*
*  This script is part of the Typo3 project. The Typo3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * $Id$
 *
 */

/**
  * This class is a demo widget for t3blog. It embeds Google trends to the page.
  *
  * @author Dmitry Dulepov <dmitry.dulepov@gmail.com>
  * @package TYPO3
  * @subpackage tx_t3blogwidgetdemo
  */

class tx_t3blogwidgetdemo_trend {

	/**
	 * Produces the output.
	 *
	 * @param string $unused
	 * @param array $conf
	 * @param array $piVars
	 * @param tslib_cObj $cObj
	 */
	public function main($unused, array $conf, $piVars, tslib_cObj $cObj) {
		$html = '<iframe src="http://www.gmodules.com/ig/ifr?url=' .
			'http://www.google.com/ig/modules/trends_gadget.xml&amp;' .
			'source=imag&amp;up_is_init=true&amp;up_cur_term=' .
			htmlspecialchars(rawurlencode($conf['terms'])) .
			'&amp;up_date=all&amp;up_region=all" ' .
			'style="border:1px solid #ccc; padding:10px;" ' .
			'width="' . htmlspecialchars($conf['width']) .
			'" height="' . htmlspecialchars($conf['height']) .
			'" frameborder="0" scrolling="no"></iframe>';

		return $html;
	}
}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/t3blog_widgetdemo/class.tx_t3blogwidgetdemo_trend.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/t3blog_widgetdemo/class.tx_t3blogwidgetdemo_trend.php']);
}

?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "t3blog_widgetdemo".
#
# Auto generated 27-05-2010 15:06
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'T3BLOG: widget demo',
	'description' => 'This extension shows how to create extra widgets for t3blog extension. See README.txt inside the extension.',
	'category' => 'fe',
	'author' => 'Dmitry Dulepov',
	'author_email' => 'dmitry.dulepov@gmail.com',
	'shy' => '',
	'dependencies' => 't3blog',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'beta',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => 'Snowflake Productions Gmbh',
	'version' => '0.1.0',
	'constraints' => array(
		'depends' => array(
			't3blog' => '0.9.0-',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:9:{s:9:"ChangeLog";s:4:"3b50";s:10:"README.txt";s:4:"3643";s:12:"ext_icon.gif";s:4:"1bdc";s:17:"ext_localconf.php";s:4:"5785";s:14:"ext_tables.php";s:4:"612a";s:16:"static/setup.txt";s:4:"1084";s:48:"widgets/class.tx_t3blogwidgetdemo_getwidgets.php";s:4:"d5d4";s:49:"widgets/trend/class.tx_t3blogwidgetdemo_trend.php";s:4:"068d";s:27:"widgets/trend/locallang.xml";s:4:"8b3b";}',
	'suggests' => array(
	),
);

?>